scripts/globals/job_utils/ranger.lua

```diff
@@ -0,0 +1,284 @@
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkDecoyShot = function(player, target, ability)
+    return 0, 0
+end
+
+xi.job_utils.ranger.checkHoverShot = function(player, target, ability)
+    return 0, 0  -- Not implemented yet
+end
+
+xi.job_utils.ranger.checkOverkill = function(player, target, ability)
+    ability:setRecast(math.max(0, ability:getRecast() - player:getMod(xi.mod.ONE_HOUR_RECAST) * 60))
+
+    return 0, 0
+end
+
+-----------------------------------
+-- Ability Use Functions
+-----------------------------------
+
+xi.job_utils.ranger.useEagleEyeShot = function(player, target, ability)
+ if player:getWeaponSkillType(xi.slot.RANGED) == xi.skill.MARKSMANSHIP then
+        action:setAnimation(target:getID(), action:getAnimation(target:getID()) + 1)
+    end
+
+    local params = {}
+
+    params.numHits = 1
+
+    -- TP params.
+    local tp          = 1000 -- to ensure ftp multiplier is applied
+    params.ftpMod     = { 5.0, 5.0, 5.0 }
+    params.critVaries = { 0.0, 0.0, 0.0 }
+
+    -- Stat params.
+    params.str_wsc = 0
+    params.dex_wsc = 0
+    params.vit_wsc = 0
+    params.agi_wsc = 0
+    params.int_wsc = 0
+    params.mnd_wsc = 0
+    params.chr_wsc = 0
+
+    params.enmityMult = 0.5
+
+    -- Job Point Bonus Damage
+    local jpValue = player:getJobPointLevel(xi.jp.EAGLE_EYE_SHOT_EFFECT)
+    player:addMod(xi.mod.ALL_WSDMG_ALL_HITS, jpValue * 3)
+
+    local damage, _, tpHits, extraHits = xi.weaponskills.doRangedWeaponskill(player, target, 0, params, tp, action, true)
+
+    -- Set the message id ourselves
+    if tpHits + extraHits > 0 then
+        action:messageID(target:getID(), xi.msg.basic.JA_DAMAGE)
+        action:speceffect(target:getID(), 32)
+    else
+        action:messageID(target:getID(), xi.msg.basic.JA_MISS_2)
+        action:speceffect(target:getID(), 0)
+    end
+
+    return damage
+end
+
+xi.job_utils.ranger.useVelocityShot = function(player, target, ability)
+	player:addStatusEffect(xi.effect.VELOCITY_SHOT, 1, 0, 7200)
+end
+
+xi.job_utils.ranger.useSharpshot = function(player, target, ability)
+    local power = 40 + player:getMod(xi.mod.SHARPSHOT)
+    player:addStatusEffect(xi.effect.SHARPSHOT, power, 0, 60)
+end
+
+xi.job_utils.ranger.useScavenge = function(player, target, ability)
+    -- RNG AF2 quest check
+    local fireAndBrimstoneCS = player:getCharVar('fireAndBrimstone')
+    local oldEarring         = 1113 -- TODO: Use items file.
+
+    if
+        player:getZoneID() == 151 and fireAndBrimstoneCS == 5 and-- zone + quest match
+        not player:hasItem(oldEarring) and -- make sure player doesn't already have the earring
+        player:getYPos() > -43 and player:getYPos() < -38 and -- Y match
+        player:getXPos() > -85 and player:getXPos() < -73 and -- X match
+        player:getZPos() > -85 and player:getZPos() < -75 and -- Z match
+        math.random(1, 100) <= 50
+    then
+        if player:getFreeSlotsCount() == 0 then
```

Comment:
This entire condition can be handled by npcUtil
`npcUtil.giveItem(player, xi.item.OLD_EARRING)`

Reply:
checked the npcUtils and are you referring to the conditions in 175 -> 181 or also the 169-170 ? I am guessing the pos check is so we are aware that the ranger is in the lil pool in the castle with the leeches while they are doing their job ability :)

Reply:
however, im wondering if that getXPos... check could be simplified by just creating a bounding box and checking if the player pos is within that instead of checking each coordinate separately

Reply:
fixed in latest commits referring to all logic after then